### PR TITLE
chore: generate networking.k8s.io/v1 version of Ingress

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Generates networking.k8s.io/v1 version of Ingress

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20182

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: bundle/next/eclipse-che-preview-openshift/manifests/che-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -536,6 +536,16 @@ spec:
                 - watch
                 - get
                 - delete
+            - apiGroups:
```
Comment (mshaposhnik) [inline bundle/next/eclipse-che-preview-openshift/manifests/che-operator.clusterserviceversion.yaml:539]: Pls add into workspace-namespaces as well, thanks

File: bundle/next/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1120,6 +1130,12 @@ spec:
                 - ingresses
               verbs:
                 - '*'
+            - apiGroups:
+                - networking.k8s.io
+              resources:
+                - ingresses
+              verbs:
+                - '*'
```
Comment (amisevsk) [inline bundle/next/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml:1138]: This supercedes the permission above (line 473--482)

File: controllers/checlusterrestore/backup_data_restorer.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/sirupsen/logrus"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	networking "k8s.io/api/networking/v1"
```
Comment (amisevsk) [inline controllers/checlusterrestore/backup_data_restorer.go:31]: To match the other names
```suggestion
	networkingv1 "k8s.io/api/networking/v1"
```

File: pkg/deploy/ingres_test.go [hunk 4]
```diff
@@ -18,11 +18,11 @@ import (
 	"github.com/google/go-cmp/cmp"
 
 	orgv1 "github.com/eclipse-che/che-operator/api/v1"
-	"k8s.io/api/extensions/v1beta1"
+	networking "k8s.io/api/networking/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
```
Comment (amisevsk) [inline pkg/deploy/ingres_test.go:25]: Is this required?

**Review by mmorhun — APPROVED** [review]

File: bundle/next/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (tolusha) [reply bundle/next/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml:1138]: The first one is is cluster roles another one just roles.

**Pull request merged**